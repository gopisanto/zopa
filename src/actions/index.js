// import axios from 'axios';

export const ADD_TRANSACTION = 'src.actions.addTransaction';

export const addTransaction = payload => async (dispatch, getState) => {
  dispatch({
    type: ADD_TRANSACTION,
    payload,
  });
};
