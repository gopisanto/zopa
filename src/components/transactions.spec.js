import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import Transactions from './transactions';
import reducer from '../reducers';
import { ADD_TRANSACTION } from '../actions';

Enzyme.configure({ adapter: new Adapter() });

let store;

describe('Transactions component', () => {
  beforeEach(() => {
    store = createStore(reducer);
  });

  test('should render nothing when no transactions', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Transactions />
      </Provider>
    );
    expect(wrapper.find('.lbl-no-transactions').text()).toEqual('No transactions.');
  });

  test('should render same snapshot when no transactions', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Transactions />
      </Provider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  test('should render transactions', () => {
    const transactions = [
      {
        name: 'ZOPA',
        email: 'zopa@gmail.com',
        amountTransferred: 1000,
      },
      {
        name: 'ZOPA1',
        email: 'zopa@zopa.de',
        amountTransferred: 4500,
      }
    ];

    store.dispatch({
      type: ADD_TRANSACTION,
      payload: transactions[0],
    });

    store.dispatch({
      type: ADD_TRANSACTION,
      payload: transactions[1],
    });

    const wrapper = mount(
      <Provider store={store}>
        <Transactions />
      </Provider>
    );

    expect(wrapper.find('.transaction').length).toEqual(transactions.length);
  });

  test('should render same snapshot when some transactions', () => {
    const transactions = [
      {
        name: 'ZOPA',
        email: 'zopa@gmail.com',
        amountTransferred: 1000,
      },
      {
        name: 'ZOPA1',
        email: 'zopa@zopa.de',
        amountTransferred: 4500,
      }
    ];

    store.dispatch({
      type: ADD_TRANSACTION,
      payload: transactions[0],
    });

    store.dispatch({
      type: ADD_TRANSACTION,
      payload: transactions[1],
    });

    const component = renderer.create(
      <Provider store={store}>
        <Transactions />
      </Provider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
