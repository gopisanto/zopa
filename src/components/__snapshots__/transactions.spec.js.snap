// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions component should render same snapshot when no transactions 1`] = `
<div
  className="transactions"
>
  <h1>
    My account
  </h1>
  <div
    className="overview"
  >
    <span>
      <label>
        £0.0000
      </label>
      <label>
        total sent
      </label>
    </span>
    <div
      className="account-chart"
    >
      <svg
        viewBox="0 0 42 42"
      >
        <circle
          cx={21}
          cy={21}
          fill="none"
          r={15.915494309189533}
          stroke="#FDB424"
          strokeDasharray="100 0"
          strokeDashoffset={100}
          strokeWidth={10}
        />
      </svg>
    </div>
    <span>
      <label>
        £100,000.0000
      </label>
      <label>
        left available
      </label>
    </span>
  </div>
  <div
    className="detail"
  >
    <h3>
      Transactions
    </h3>
    <label
      className="lbl-no-transactions"
    >
      No transactions.
    </label>
  </div>
</div>
`;

exports[`Transactions component should render same snapshot when some transactions 1`] = `
<div
  className="transactions"
>
  <h1>
    My account
  </h1>
  <div
    className="overview"
  >
    <span>
      <label>
        £5,500.0000
      </label>
      <label>
        total sent
      </label>
    </span>
    <div
      className="account-chart"
    >
      <svg
        viewBox="0 0 42 42"
      >
        <circle
          cx={21}
          cy={21}
          fill="none"
          r={15.915494309189533}
          stroke="#ECE3EB"
          strokeDasharray="5.5 94.5"
          strokeDashoffset={0}
          strokeWidth={10}
        />
        <circle
          cx={21}
          cy={21}
          fill="none"
          r={15.915494309189533}
          stroke="#FDB424"
          strokeDasharray="94.5 5.5"
          strokeDashoffset={94.5}
          strokeWidth={10}
        />
      </svg>
    </div>
    <span>
      <label>
        £94,500.0000
      </label>
      <label>
        left available
      </label>
    </span>
  </div>
  <div
    className="detail"
  >
    <h3>
      Transactions
    </h3>
    <div
      className="transaction"
    >
      <div
        className="person"
      >
        <label
          className="name"
        >
          ZOPA
        </label>
        <label
          className="email"
        >
          zopa@gmail.com
        </label>
      </div>
      <div
        className="amount"
      >
        <label>
          £1,000.0000
        </label>
      </div>
    </div>
    <div
      className="transaction"
    >
      <div
        className="person"
      >
        <label
          className="name"
        >
          ZOPA1
        </label>
        <label
          className="email"
        >
          zopa@zopa.de
        </label>
      </div>
      <div
        className="amount"
      >
        <label>
          £4,500.0000
        </label>
      </div>
    </div>
  </div>
</div>
`;
